src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC.js

```diff
@@ -0,0 +1,257 @@
+        this._previousApMasterStatus = false;
+        this._apMasterStatus = false;
+        this._apHasDeactivated = false;
+        this._hasReachedTopOfDescent = false;
+        this._apCooldown = 500;
+    }
+    get templateID() { return "CJ4_FMC"; }
+    connectedCallback() {
+        super.connectedCallback();
+        this.radioNav.init(NavMode.TWO_SLOTS);
+        if (!this._registered) {
+            RegisterViewListener("JS_LISTENER_KEYEVENT", () => {
+                console.log("JS_LISTENER_KEYEVENT registered.");
+                RegisterViewListener("JS_LISTENER_FACILITY", () => {
+                    console.log("JS_LISTENER_FACILITY registered.");
+                    this._registered = true;
+                });
+            });
+        }
+    }
+    Init() {
+        super.Init();
+        this.maxCruiseFL = 450;
+        this.onFplan = () => { CJ4_FMC_RoutePage.ShowPage1(this); };
+        this.onLegs = () => { CJ4_FMC_LegsPage.ShowPage1(this); };
+        this.onDepArr = () => { CJ4_FMC_DepArrPage.ShowPage1(this); };
+        this.onDsplMenu = () => { CJ4_FMC_InitRefIndexPage.ShowPage1(this); };
+        this.onTun = () => { CJ4_FMC_NavRadioPage.ShowPage1(this); };
+        this.onExec = () => {
+            if (this.getIsRouteActivated()) {
+                this.insertTemporaryFlightPlan();
+                this._isRouteActivated = false;
+                SimVar.SetSimVarValue("L:FMC_EXEC_ACTIVE", "number", 0);
+                if (this.refreshPageCallback) {
+                    this.refreshPageCallback();
+                }
+            }
+        };
+        CJ4_FMC_IdentPage.ShowPage1(this);
+        
+        //Timer for periodic page refresh
+        this._pageRefreshTimer = null;
+    }
+    Update() {
+        super.Update();
+        this.updateAutopilot();
+    }
+    onInputAircraftSpecific(input) {
+        console.log("CJ4_FMC.onInputAircraftSpecific input = '" + input + "'");
+        if (input === "LEGS") {
+            if (this.onLegs) {
+                this.onLegs();
+            }
+            return true;
+        }
+        if (input === "FPLN") {
+            if (this.onFplan) {
+                this.onFplan();
+            }
+            return true;
+        }
+        if (input === "DSPL_MENU") {
+            if (this.onDsplMenu) {
+                this.onDsplMenu();
+            }
+            return true;
+        }
+        if (input === "TUN") {
+            if (this.onTun) {
+                this.onTun();
+            }
+            return true;
+        }
+        if (input === "DIR") {
+            CJ4_FMC_DirectToPage.ShowPage(this);
+        }
+        if (input === "EXEC") {
+            if (this.onExec) {
+                this.onExec();
+            }
+        }
+        return false;
+    }
+    clearDisplay() {
+        super.clearDisplay();
+        this.onPrevPage = EmptyCallback.Void;
+        this.onNextPage = EmptyCallback.Void;
+
+        this.unregisterPeriodPageRefresh();
```

Comment:
Any time anyone calls for the display to be cleared, we unregister any periodic refresh. Else everyone's intervals would be running on top of each other and getting recreated but never removed.